src/main/java/app/gpx_animator/Renderer.java

```diff
@@ -547,14 +572,83 @@ private void drawInfo(final BufferedImage bi, final int frame, final Point2D mar
+            cfg.getMargin() + fontMetrics.getHeight() * 2);
+            printText(graphics, latLongString, (float) (bi.getWidth() - fontMetrics.stringWidth(latLongString)) / 2,
+            cfg.getMargin() + fontMetrics.getHeight());
+            printText(graphics, speedString, (float) (bi.getWidth() - fontMetrics.stringWidth(speedString)) / 2, cfg.getMargin());
+        } else if (informationPosition.equals(topRight)) {
+            printText(graphics, dateString, bi.getWidth() - fontMetrics.stringWidth(dateString) - cfg.getMargin(),
+            cfg.getMargin() + fontMetrics.getHeight() * 2);
+            printText(graphics, latLongString, bi.getWidth() - fontMetrics.stringWidth(latLongString) - cfg.getMargin(),
+            cfg.getMargin() + fontMetrics.getHeight());
+            printText(graphics, speedString, bi.getWidth() - fontMetrics.stringWidth(speedString) - cfg.getMargin(), cfg.getMargin());
+        } else if (informationPosition.equals(bottomLeft)) {
+            printText(graphics, dateString, cfg.getMargin(), bi.getHeight() - cfg.getMargin());
+            printText(graphics, latLongString, cfg.getMargin(), bi.getHeight() - cfg.getMargin() - fontMetrics.getHeight());
+            printText(graphics, speedString, cfg.getMargin(), bi.getHeight() - cfg.getMargin() - fontMetrics.getHeight() * 2);
+        } else if (informationPosition.equals(bottomCenter)) {
+            printText(graphics, dateString, (float) (bi.getWidth() - fontMetrics.stringWidth(dateString)) / 2, bi.getHeight() - cfg.getMargin());
+            printText(graphics, latLongString, (float) (bi.getWidth() - fontMetrics.stringWidth(latLongString)) / 2,
+            bi.getHeight() - cfg.getMargin() - fontMetrics.getHeight());
+            printText(graphics, speedString, (float) (bi.getWidth() - fontMetrics.stringWidth(speedString)) / 2,
+            bi.getHeight() - cfg.getMargin() - fontMetrics.getHeight() * 2);
+        } else if (informationPosition.equals(bottomRight)) {
+            printText(graphics, dateString, bi.getWidth() - fontMetrics.stringWidth(dateString) - cfg.getMargin(), bi.getHeight() - cfg.getMargin());
+            printText(graphics, latLongString, bi.getWidth() - fontMetrics.stringWidth(latLongString) - cfg.getMargin(),
+            bi.getHeight() - cfg.getMargin() - fontMetrics.getHeight());
+            printText(graphics, speedString, bi.getWidth() - fontMetrics.stringWidth(speedString) - cfg.getMargin(),
+            bi.getHeight() - cfg.getMargin() - fontMetrics.getHeight() * 2);
+        }
+    }
+
+private String getSpeedString(final Point2D point, final int frame) {
+        if (point instanceof GpxPoint) {
+            final GpxPoint gpxPoint = (GpxPoint) point;
+            final long speed = calculateSpeedForDisplay(gpxPoint, frame);
+            final String calculation = calcToMinPer(speed);
+            return calculation;
+        } else {
+            return "";
+        }
```

Comment:
I moved this method to the `SpeedUtil` class. Just remove it here and it will be fine.